include/rmm/device_scalar.hpp

```diff
@@ -75,12 +106,12 @@ class device_scalar {
    * @param mr Optional, resource with which to allocate.
    */
   explicit device_scalar(
-    T const &initial_value,
-    cuda_stream_view stream             = cuda_stream_view{},
+    value_type const &initial_value,
+    cuda_stream_view stream,
```

Comment:
Why remove the default? It seems that it would be a common use case to use the default stream, especially with PTDS enabled.

Reply:
Because we want to make all stream-ordered APIs explicit. See e.g. #418.  Note that `device_uvector` and now `device_buffer` have no default stream parameters.